modules/sfp_opencorporates.py

```diff
@@ -0,0 +1,131 @@
+# -*- coding: utf-8 -*-
+# -------------------------------------------------------------------------------
+# Name:         sfp_opencorporates
+# Purpose:      SpiderFoot plug-in for retrieving company location and previous
+#               company names from OpenCorporates.
+#
+# Author:      Brendan Coles <bcoles@gmail.com>
+#
+# Created:     2018-10-21
+# Copyright:   (c) Brendan Coles 2018
+# Licence:     GPL
+# -------------------------------------------------------------------------------
+
+import json
+import re
+import urllib
+from sflib import SpiderFoot, SpiderFootPlugin, SpiderFootEvent
+
+class sfp_opencorporates(SpiderFootPlugin):
+    """OpenCorporates:Passive,Footprint,Investigate:Search Engines::Look up company information from OpenCorporates."""
+
+    # Default options
+    opts = {
+        'confidence': 100
```

Comment:
What about the ability for the user to enter in their own API key?

Reply:
Updated to support API keys